bugmon.py

```diff
@@ -15,1092 +15,489 @@
+                elif op_sys.find('Mac OS'):
+                    self._os = 'Darwin'
+                else:
+                    self._os = platform.system()
             else:
-              bisectFixComments.append("JSBugMon: Fix Bisection requested, failed due to error (try manually).")
-              bisectFixComments.append("");
-          else:
-            # Threat this as a temporary failure, don't remove the whiteboard tag
-            bugBisectFixRequested = False
-
-      wbParts = []
-      whiteBoardModified = False
-      if closeBug or verifyBug or len(comments) > 0:
-        whiteBoardModified = True
-        wbOpts.append('ignore')
-
-      if bugBisectRequested:
-        whiteBoardModified = True
-        wbOpts.remove('bisect')
-        comments.extend(bisectComments)
-
-      if bugBisectFixRequested and len(bisectFixComments) > 0:
-        whiteBoardModified = True
-        wbOpts.remove('bisectfix')
-        comments.extend(bisectFixComments)
-
-      if ((bugBisectRequested or (bugBisectFixRequested and len(bisectFixComments) > 0)) and bugBisectForceCompile):
-        whiteBoardModified = True
-        wbOpts.remove('bisect-force-compile')
-        
-      if bugFailureMsg != None and bugUpdateRequested:
-        whiteBoardModified = True
-        wbOpts.remove('update')
-        comments.append(bugFailureMsg)
-
-      if whiteBoardModified:
-        wbParts = filter(lambda x: len(x) > 0, map(str.rstrip, map(str.lstrip, re.split('\[jsbugmon:[^\]]+\]', bug.whiteboard))))
-        wbParts.append("[jsbugmon:" + ",".join(wbOpts) + "]")
-
-      while True:
-        # Fetch the bug again
-        bug = self.fetchBug(bugnum)
-
-        bugModified = False
-
-        # Mark bug as WORKSFORME if confirmed to no longer reproduce
-        if closeBug:
-          bugModified = True
-          bug.status = "RESOLVED"
-          bug.resolution = "WORKSFORME"
-
-        # Mark bug as VERIFIED if we verified it successfully
-        if verifyBug:
-          bugModified = True
-          bug.status = "VERIFIED"
-
-        if whiteBoardModified:
-          # We add "ignore" to our bugmon options so we don't update the bug a second time
-          bugModified = True
-          bug.whiteboard = " ".join(wbParts)
-        
+                self._os = platform.system()
+
+        return self._os
+
+    @property
+    def arch(self):
+        """
+        Attempt to enumerate the original architecture associated with the bug
+        """
+        # ToDo: This does nothing
+        #   This needs to be modified to check if the specified bug platform matches our local platform
+        #   If not, error out
+        # if self._arch is None:
+        #     if self.bug.platform == 'ARM':
+        #         return 'ARM64'
+        #     elif self.bug.platform == 'x86':
+        #         return 'i686'
+        #     elif self.bug.platform == 'x86_64':
+        #         return 'AMD64'
+
+        return platform.machine()
```

Comment:
I don't understand this. The platform is not just relevant for checking, but (for the JS engine) also used for selecting the right build. We do have x86 and ARM/ARM64 (simulator) builds on x86_64. For the JS engine we should probably try to use these. Alternatively, we can of course run Bugmon on each architecture natively.